### PR TITLE
Factor out `hir::Node::Binding`

### PR DESCRIPTION
**Pull request by camsteffen**

### CODE REVIEW

File: src/tools/clippy/clippy_lints/src/escape.rs [hunk 1]
```diff
@@ -143,15 +146,6 @@ impl<'a, 'tcx> Delegate<'tcx> for EscapeDelegate<'a, 'tcx> {
         if cmt.place.projections.is_empty() {
             if let PlaceBase::Local(lid) = cmt.place.base {
                 self.set.remove(&lid);
-                let map = &self.cx.tcx.hir();
-                if let Some(Node::Binding(_)) = map.find(cmt.hir_id) {
-                    if self.set.contains(&lid) {
```
Comment (camsteffen) [inline src/tools/clippy/clippy_lints/src/escape.rs:148]: Dead code. Was just removed!

**Comment by camsteffen** [comment]
My bad. Fixed tidy and rebased. Didn't run the ui tests since I'm hoping they just pass.

**Comment by compiler-errors** [comment]
I can take a look at this after my real day job is over :stuck_out_tongue_winking_eye:.

**Comment by compiler-errors** [comment]
Looks good to me.

@bors r+

**Comment by matthiaskrgr** [comment]
This will need a rebase once https://github.com/rust-lang/rust/pull/98706 is merged, can't merge both PRs at a time due to compiler error: https://github.com/rust-lang/rust/pull/98723#issuecomment-1171440249

I'm gonna @bors r- to get this out of the queue for now, feel free to re-r+ once rebase is done.

File: src/tools/clippy/clippy_lints/src/manual_rem_euclid.rs [hunk 2]
```diff
@@ -71,8 +71,7 @@ impl<'tcx> LateLintPass<'tcx> for ManualRemEuclid {
             && let Some(const3) = check_for_unsigned_int_constant(cx, right)
             // Also ensures the const is nonzero since zero can't be a divisor
             && const1 == const2 && const2 == const3
-            && let Some(hir_id) = path_to_local(expr3)
-            && let Some(Node::Binding(_)) = cx.tcx.hir().find(hir_id) {
+            && let Some(hir_id) = path_to_local(expr3) {
```
Comment (camsteffen) [inline src/tools/clippy/clippy_lints/src/manual_rem_euclid.rs:74]: New change with rebase. `path_to_local` is always a binding.

**Pull request merged**